db/schema.rb

```diff
@@ -196,10 +198,50 @@
     t.integer "option_id", null: false, index: {name: "fk__course_assessment_multiple_response_option_question_option"}, foreign_key: {references: "course_assessment_question_multiple_response_options", name: "fk_course_assessment_answer_multiple_response_options_option_id", on_update: :no_action, on_delete: :no_action}
   end
 
+  create_table "course_assessment_answer_programming", force: :cascade do |t|
+  end
+
+  create_table "course_assessment_answer_programming_auto_gradings", force: :cascade do |t|
+  end
+
+  create_table "course_assessment_question_programming", force: :cascade do |t|
+    t.string  "language",      limit: 255, null: false
+    t.integer "memory_limit",  comment: "Memory limit, in MiB"
+    t.integer "time_limit",    comment: "Time limit, in seconds"
+    t.uuid    "import_job_id", comment: "The ID of the importing job", index: {name: "index_course_assessment_question_programming_on_import_job_id", unique: true}, foreign_key: {references: "jobs", name: "fk_course_assessment_question_programming_import_job_id", on_update: :no_action, on_delete: :nullify}
+  end
+
+  create_table "course_assessment_question_programming_test_cases", force: :cascade do |t|
+    t.integer "question_id", null: false, index: {name: "fk__course_assessment_quest_18b37224652fc59d955122a17ba20d07"}, foreign_key: {references: "course_assessment_question_programming", name: "fk_course_assessment_questi_ee00a2daf4389c4c2ddba3041a15c35f", on_update: :no_action, on_delete: :no_action}
+    t.string  "identifier",  limit: 255, null: false, comment: "Test case identifier generated by the testing framework", index: {name: "index_course_assessment_question_programming_test_case_ident", with: ["question_id"], unique: true}
+    t.boolean "public",      null: false
+    t.text    "description", null: false
+    t.text    "hint"
+  end
+
+  create_table "course_assessment_answer_programming_auto_grading_test_results", force: :cascade do |t|
+    t.integer "auto_grading_id", null: false, index: {name: "fk__course_assessment_answe_57b22f114b54911fd4e1519680ebfd49"}, foreign_key: {references: "course_assessment_answer_programming_auto_gradings", name: "fk_course_assessment_answer_e3d785447112439bb306849be8690102", on_update: :no_action, on_delete: :no_action}
+    t.integer "test_case_id",    index: {name: "fk__course_assessment_answe_29a2568d5e2fb3a47c0561815786f9ab"}, foreign_key: {references: "course_assessment_question_programming_test_cases", name: "fk_course_assessment_answer_bbb492885b1e3dca4433b8af8cb95906", on_update: :no_action, on_delete: :no_action}
+    t.boolean "passed",          null: false
+    t.text    "message"
+  end
+
+  create_table "course_assessment_answer_programming_files", force: :cascade do |t|
```

Comment:
So assessments will use its own file model ?  Will this do anything with the attachment in future ?